modules/login/php/requestaccount.class.inc

```diff
@@ -171,6 +181,14 @@ class RequestAccount extends \NDB_Form
                 )
             );
 
+            $DB->insert(
+                'user_project_rel',
+                array(
+                    'UserID'    => $user_id,
+                    'ProjectID' => $project,
```

Comment:
`$project` came from the value submitted by the user. It needs to be validated by the backend before being inserted and shouldn't be blindly trusted.

Reply:
then same should be done with site

Reply:
True, might as well do that at the same time.

Reply:
@driusan what kind of validation would you like to see here? The values for site and project are coming from drop-downs populated with values from the DB.

Reply:
they are then passed to the insert query as parameters and inserted into a table with foreign keys...

Reply:
1. The value submitted is an integer.
2. The value integer submitted exists as a key in the appropriate (project/psc) tables.